subsys/mgmt/osdp/src/osdp_cp.c

```diff
@@ -0,0 +1,888 @@
+			break;
+		}
+		pos++;	         /* reader number; skip */
+		t1 = buf[pos++]; /* key length */
+		if ((len - REPLY_KEYPPAD_DATA_LEN) != t1) {
+			break;
+		}
+		if (cp->notifier.keypress) {
+			for (i = 0; i < t1; i++) {
+				t2 = buf[pos + i]; /* key data */
+				cp->notifier.keypress(pd->offset, t2);
+			}
+		}
+		ret = 0;
+		break;
+	case REPLY_RAW:
+		if (len < REPLY_RAW_DATA_LEN) {
+			break;
+		}
+		pos++;	                /* reader number; skip */
+		t1 = buf[pos++];        /* format */
+		t2 = buf[pos++];        /* length LSB */
+		t2 |= buf[pos++] << 8; /* length MSB */
+		if ((len - REPLY_RAW_DATA_LEN) != t2) {
+			break;
+		}
+		if (cp->notifier.cardread) {
+			cp->notifier.cardread(pd->offset, t1, buf + pos, t2);
+		}
+		ret = 0;
+		break;
+	case REPLY_FMT:
+		if (len < REPLY_FMT_DATA_LEN) {
+			break;
+		}
+		pos++;	/* reader number; skip */
+		pos++;	/* skip one byte -- TODO: handle reader direction */
+		t1 = buf[pos++]; /* Key length */
+		if ((len - REPLY_FMT_DATA_LEN) != t1) {
+			break;
+		}
+		if (cp->notifier.cardread) {
+			cp->notifier.cardread(pd->offset, OSDP_CARD_FMT_ASCII,
+					      buf + pos, t1);
+		}
+		ret = 0;
+		break;
+	case REPLY_BUSY:
+		/* PD busy; signal upper layer to retry command */
+		if (len != REPLY_BUSY_DATA_LEN) {
+			break;
+		}
+		ret = 2;
+		break;
+	default:
+		LOG_DBG(TAG "unexpected reply: 0x%02x", pd->reply_id);
+		return -1;
+	}
+
+	if (ret == -1) {
+		LOG_ERR(TAG "REPLY %02x for CMD %02x format error!",
+			pd->cmd_id, pd->reply_id);
+		return -1;
+	}
+
+	if (pd->cmd_id != CMD_POLL) {
+		LOG_DBG(TAG "CMD: %02x REPLY: %02x", pd->cmd_id, pd->reply_id);
+	}
+
+	return ret;
+}
+
+/**
+ * cp_send_command - blocking send; doesn't handle partials
+ * Returns:
+ *   0 - success
+ *  -1 - failure
+ */
+static int cp_send_command(struct osdp_pd *pd)
+{
+	int ret, len;
+
+	/* init packet buf with header */
+	len = osdp_phy_packet_init(pd, pd->rx_buf, sizeof(pd->rx_buf));
+	if (len < 0) {
+		return -1;
+	}
+
+	/* fill command data */
+	ret = cp_build_command(pd, pd->rx_buf, sizeof(pd->rx_buf));
+	if (ret < 0) {
+		return -1;
+	}
+	len += ret;
+
+	/* finalize packet */
+	len = osdp_phy_packet_finalize(pd, pd->rx_buf, len, sizeof(pd->rx_buf));
+	if (len < 0) {
+		return -1;
+	}
+
+	ret = pd->channel.send(pd->channel.data, pd->rx_buf, len);
+
+	if (IS_ENABLED(CONFIG_OSDP_PACKET_TRACE)) {
+		if (pd->cmd_id != CMD_POLL) {
+			LOG_DBG(TAG "bytes sent");
+			osdp_dump(NULL, pd->rx_buf, len);
+		}
+	}
+
+	return (ret == len) ? 0 : -1;
+}
+
+/**
+ * cp_process_reply - received buffer from serial stream handling partials
+ * Returns:
+ *  0: success
+ * -1: error
+ *  1: no data yet
+ *  2: re-issue command
+ */
+static int cp_process_reply(struct osdp_pd *pd)
+{
+	uint8_t *buf;
+	int rec_bytes, ret, max_len;
+
+	buf = pd->rx_buf + pd->rx_buf_len;
+	max_len = sizeof(pd->rx_buf) - pd->rx_buf_len;
+
+	rec_bytes = pd->channel.recv(pd->channel.data, buf, max_len);
+	if (rec_bytes <= 0)	/* No data received */
```

Comment:
one liners need {}

Reply:
Sorry, force of habit. I'll try to be more careful. Btw, can we have checkpatch warn for this?